### PR TITLE
Unit test payment

### PR DESCRIPTION
**Pull request by zenithyap**
Added unit test for the following functions:
- braintreeTokenController
- brainTreePaymentController

### CODE REVIEW

File: controllers/productController.test.js [hunk 1]
```diff
@@ -1,545 +1,703 @@
...
+                products: req.body.cart,
+                payment: expect.any(Object),
+                buyer: req.user._id
+            });
+
+            expect(res.status).toHaveBeenCalledWith(200);
+            expect(res.json).toHaveBeenCalledWith({ ok: true });
+        });
+
+        test('should handle error when perfoming transaction sale', async () => {
```
Comment (eugenetangkj) [inline controllers/productController.test.js]: Typo with `performing`

File: controllers/productController.test.js [hunk 2]
```diff
@@ -1,545 +1,703 @@
...
+        // NEVER PASS
+        test('should handle error when thrown in try block', async () => {
+            const consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => { });
+            const error = new Error('Test error');
+            const gateway = new braintree.BraintreeGateway();
+            gateway.transaction.sale.mockImplementation(() => {
+                throw error;
+            });
+
+            await braintreeTokenController(req, res);
```
Comment (eugenetangkj) [inline controllers/productController.test.js]: Why is this `braintreeTokenController` and not `brainTreePaymentController`?

**Review by eugenetangkj — COMMENTED** [review]
Left some comments for your perusal

File: controllers/productController.test.js (hunk 1 above)
Comment (zenithyap) [reply controllers/productController.test.js]: Oops, I have changed it

File: controllers/productController.test.js (hunk 2 above)
Comment (zenithyap) [reply controllers/productController.test.js]: I have changed it to brainTreePaymentController

**Pull request merged**